storage/src/test/java/org/apache/kafka/tiered/storage/utils/RecordsKeyValueMatcher.java

```diff
@@ -138,18 +140,21 @@ private boolean compare(ByteBuffer lhs,
     private SimpleRecord convert(Object recordCandidate) {
         if (recordCandidate instanceof ProducerRecord) {
             ProducerRecord<?, ?> record = (ProducerRecord<?, ?>) recordCandidate;
+            long timestamp = record.timestamp() == null ? RecordBatch.NO_TIMESTAMP : record.timestamp();
             ByteBuffer keyBytes =
                     Utils.wrapNullable(keySerde.serializer().serialize(topicPartition.topic(), (K) record.key()));
             ByteBuffer valueBytes =
                     Utils.wrapNullable(valueSerde.serializer().serialize(topicPartition.topic(), (V) record.value()));
-            return new SimpleRecord(record.timestamp(), keyBytes, valueBytes, record.headers().toArray());
+            Header[] headers = record.headers() != null ? record.headers().toArray() : Record.EMPTY_HEADERS;
```

Comment:
nit: may introduce `getHeaders(record)` to return headers as it is used at other places. 
```
Header[] getHeaders(ProducerRecord record) {
    return record.headers() != null ? record.headers().toArray() : Record.EMPTY_HEADERS;
}
```

Reply:
We have three different records: ProducerRecord, ConsumerRecord and SimpleRecord. They don't extend a common interface.

Reply:
okay, makes sense.